packages/pages/src/upgrade/migrateConfig.ts

```diff
@@ -92,16 +92,40 @@ const migrateSiteStream = async (
     console.info(
       `migrating global data from ${siteStreamPath} to ${configYamlPath}`
     );
-    // this logic replaces $id with id and keeps id in the first position
-    const newSiteStream = {
-      id: sitesJson.$id,
-      ...sitesJson,
-    };
-    newSiteStream.$id = undefined;
+    const newSiteStream = formatSiteStream(sitesJson, siteStreamPath);
+    if (!newSiteStream) {
+      return null;
+    }
     writeYamlSync(configYamlPath, "siteStream", newSiteStream);
   }
 };
 
+const formatSiteStream = (sitesJson: any, siteStreamPath: string) => {
+  let entityId: string | undefined = undefined;
+  if (sitesJson.filter?.entityIds && sitesJson.filter?.entityIds.length === 1) {
+    entityId = sitesJson.filter.entityIds[0];
+  } else if (sitesJson.filter?.entityIds) {
+    console.error(
```

Comment:
We should use the colorized version and also kill the update script. You can use `logErrorAndExit` for that.